src/main/kotlin/app/revanced/patches/youtube/layout/returnyoutubedislike/patch/ReturnYouTubeDislikePatch.kt

```diff
@@ -137,35 +134,33 @@ class ReturnYouTubeDislikePatch : BytecodePatch(
 
         // region Hook for Short videos.
 
-        ShortsTextComponentParentFingerprint.result?.let {
-            context
-                .toMethodWalker(it.method)
-                .nextMethod(it.scanResult.patternScanResult!!.endIndex, true)
-                .getMethod().let { method ->
-                    with(method as MutableMethod) {
-                        // After walking, verify the found method is what's expected.
-                        if (returnType != ("Ljava/lang/CharSequence;") || parameterTypes.size != 1)
-                            return PatchResultError("Method signature did not match: $this $parameterTypes")
-
-                        val insertIndex = implementation!!.instructions.size - 1
-                        val spannedParameterRegister = instruction<OneRegisterInstruction>(insertIndex).registerA
-                        val parameter = instruction<BuilderInstruction35c>(insertIndex - 2).reference
-
-                        if (!parameter.toString().endsWith("Landroid/text/Spanned;"))
-                            return PatchResultError("Method signature parameter did not match: $parameter")
-
-                        insertShorts(insertIndex, spannedParameterRegister)
-                    }
-                }
-
-            // Additional hook, called after user dislikes.
-            with(it.mutableMethod) {
-                val insertIndex = it.scanResult.patternScanResult!!.startIndex + 2
-                val overwriteRegister = (implementation!!.instructions.elementAt(insertIndex - 1)
-                        as OneRegisterInstruction).registerA
-                insertShorts(insertIndex, overwriteRegister)
+        ShortsTextViewFingerprint.result?.let {
+            it.mutableMethod.apply {
+                val patternResult = it.scanResult.patternScanResult!!
+
+                // Boolean field that indicates if the TextView is for a dislikes button
+                val isLikesBooleanReference = instruction<ReferenceInstruction>(patternResult.endIndex).reference
+
+                val textViewFieldReference = // Like/Dislike button TextView field
+                    instruction<ReferenceInstruction>(patternResult.endIndex - 2).reference
+
+                // insert after call to parent class constructor and after a null check
```

Comment:
```suggestion
                // Check if the hooked TextView object is that of the dislike button. 
                // If RYD is disabled, or the TextView object is not that of the dislike button, the execution flow is not interrupted. 
                // Otherwise, the TextView object is modified, and the execution flow is interrupted to prevent it from being changed afterward.
```